### PR TITLE
Fix release event's double click is not correctly set on Windows

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fix the windows part of https://github.com/godotengine/godot/issues/88872 , not sure whether it's the best way to fix this, but did some test and it looks ok.

### CODE REVIEW

File: platform/windows/display_server_windows.h [hunk 1]
```diff
@@ -468,6 +468,7 @@ class DisplayServerWindows : public DisplayServer {
 	bool use_raw_input = false;
 	bool drop_events = false;
 	bool in_dispatch_input_event = false;
+	bool last_is_double_click = false;
```
Comment (Sauermann) [inline platform/windows/display_server_windows.h:471]: A boolean is not enough.

Take the following sequence of mouse buttons:
1. Right MB down
2. Wait a few seconds and execute the following steps in quick succession
3. Left MB down
4. Left MB up
5. Left MB down
6. Right MB up

In step 6 the right mouse button up input event would be wrongly considered as a double click.

So this variable should rather store the mouse button index (see [`enum class MouseButton`](https://github.com/godotengine/godot/blob/7d2ca2d8ac49cde9767e00b70f9eaf1920eb266d/core/input/input_enums.h#L111))

**Review by Sauermann — DISMISSED** [review]
Please note, that this PR only addresses the linked bug report for Windows, but not for Linux. It is fine for this PR to solve it only on Windows, but the OP should be updated and the "Fixes" Github-keyword removed.

**Comment by AThousandShips** [comment]
Adding to the above, no OS has this functionality in Godot, even when it's directly supported like macOS, so should be done everywhere at once IMO if we consider it a missing feature, or well introduce an inconsistency that will make the feature useless unless you only target Windows

File: platform/windows/display_server_windows.h (hunk 1 above)
Comment (Sauermann) [reply platform/windows/display_server_windows.h:471]: For reference: #89018
That PR keeps also the double-click state of different buttons separate by using `MouseButtonMask`.

Comment (jsjtxietian) [reply platform/windows/display_server_windows.h:471]: > In step 6 the right mouse button up input event would be wrongly considered as a double click.

Thanks! I have followed the instructions provided and reproduced this bug.

**Comment by akien-mga** [comment]
Superseded by #92582.

**Pull request closed without merging**